jobs/query_scripts/seqr_loader.py

```diff
@@ -0,0 +1,429 @@
+
+    ht = ht.annotate(
+        info=ht.info.annotate(
+            AS_VQSLOD=_fix_inf(ht.info.AS_VQSLOD),
+            AS_MQ=_fix_inf(ht.info.AS_MQ),
+            InbreedingCoeff=_fix_inf(ht.info.InbreedingCoeff),
+            AS_InbreedingCoeff=_fix_inf(ht.info.AS_InbreedingCoeff),
+        )
+    )
+    return ht
+
+
+def annotate_cohort(
+    vcf_path,
+    site_only_vqsr_vcf_path,
+    vep_ht_path,
+    out_mt_path,
+    overwrite=False,
+    genome_build='GRCh38',
+    sequencing_type='',
+    checkpoint_prefix=None,
+):
+    """
+    Convert VCF to mt, annotate for seqr loader, add VEP annotations.
+    """
+
+    def _read(path):
+        if path.strip('/').endswith('.ht'):
+            t = hl.read_table(str(path))
+        else:
+            assert path.strip('/').endswith('.mt')
+            t = hl.read_matrix_table(str(path))
+        logging.info(f'Read checkpoint {path}')
+        return t
+
+    def _checkpoint(t, file_name):
+        if checkpoint_prefix:
+            path = os.path.join(checkpoint_prefix, file_name)
+            if not overwrite and hl.hadoop_exists(os.path.join(path, '_SUCCESS')):
+                t = _read(str(path))
+            else:
+                t.write(str(path), overwrite=True)
+                logging.info(f'Wrote checkpoint {path}')
+                t = _read(str(path))
+        return t
+
+    mt = hl.import_vcf(
+        str(vcf_path),
+        reference_genome=genome_build,
+        skip_invalid_loci=True,
+        force_bgz=True,
+    )
+    logging.info(
+        f'Importing VCF {vcf_path}, '
+        f'adding VQSR annotations from {site_only_vqsr_vcf_path}, '
+        f'adding VEP annotations from {vep_ht_path}'
+    )
+
+    logging.info(f'Loading VEP Table from {vep_ht_path}')
+    # Annotate VEP. Do ti before splitting multi, because we run VEP on unsplit VCF,
+    # and hl.split_multi_hts can handle multiallelic VEP field.
+    vep_ht = hl.read_table(str(vep_ht_path))
+    logging.info(f'Adding VEP annotations into the Matrix Table from {vep_ht_path}')
+    mt = mt.annotate_rows(vep=vep_ht[mt.locus].vep)
+
+    # Splitting multi-allelics. We do not handle AS info fields here - we handle
+    # them when loading VQSR instead, and populate entrie "info" from VQSR.
+    mt = hl.split_multi_hts(
+        mt.annotate_rows(locus_old=mt.locus, alleles_old=mt.alleles)
+    )
+    mt = _checkpoint(mt, 'mt-vep-split.mt')
+
+    vqsr_ht = load_vqsr(site_only_vqsr_vcf_path, genome_build)
+    vqsr_ht = _checkpoint(vqsr_ht, 'vqsr.ht')
+
+    logging.info('Adding VQSR annotations into the Matrix Table')
+    mt = mt.annotate_globals(**vqsr_ht.index_globals())
+    mt = mt.annotate_rows(
+        # vqsr_ht has info annotation split by allele, plus the new AS-VQSR annotations
+        info=vqsr_ht[mt.row_key].info,
+        filters=mt.filters.union(vqsr_ht[mt.row_key].filters).filter(
+            lambda val: val != 'PASS'
+        ),
+    )
+    mt = _checkpoint(mt, 'mt-vep-split-vqsr.mt')
```

Comment:
how do you feel about making the VQSR load/annotation an optional action, so that the same code can be shared by AIP and cpg_utils?

I think this could be very simple if `site_only_vqsr_vcf_path` is optional, and [this code](https://github.com/populationgenomics/automated-interpretation-pipeline/blob/main/reanalysis/annotation.py#L112-L113) is added here (for non-GATK VCFs to be annotated)

Reply:
Happy to have a go at this if you think it's possible, it would allow removal of redundancy between this and AIP

Reply:
Yes, I'm all for sharing code with AIP, just with all refactoring always kept pushing this back. I'll do that in this PR :)